js/src/components/contact-information/phone-number-card/verification-code-control.js

```diff
@@ -0,0 +1,136 @@
+/**
+ * External dependencies
+ */
+import { useState, useEffect, useRef } from '@wordpress/element';
+import { Flex } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import AppInputControl from '.~/components/app-input-control';
+import './verification-code-control.scss';
+
+const KEY_CODE_LEFT = 37;
+const KEY_CODE_RIGHT = 39;
+const KEY_CODE_BACKSPACE = 8;
+
+const DIGIT_LENGTH = 6;
+const initDigits = Array( DIGIT_LENGTH ).fill( '' );
+
+const toCallbackData = ( digits ) => {
+	const code = digits.join( '' );
+	const isFilled = code.length === DIGIT_LENGTH;
+	return { code, isFilled };
+};
+
+/**
+ * @callback onCodeChange
+ * @param {Object} verification Data payload.
+ * @param {string} verification.code The current entered verification code.
+ * @param {boolean} verification.isFilled Whether all digits of validation code are filled.
+ */
+
+/**
+ * Renders a row of input elements for entering six-digit verification code.
+ *
+ * @param {Object} props React props.
+ * @param {onCodeChange} props.onCodeChange Called when the verification code are changed.
+ * @param {string} [props.resetNeedle=''] When the passed value changes, it will trigger internal state resetting for this component.
+ */
+export default function VerificationCodeControl( {
+	onCodeChange,
+	resetNeedle = '',
+} ) {
+	const inputsRef = useRef( [] );
+	const cursorRef = useRef( 0 );
+	const onCodeChangeRef = useRef();
+	const [ digits, setDigits ] = useState( initDigits );
+
+	onCodeChangeRef.current = onCodeChange;
+
+	const maybeMoveFocus = ( targetIdx ) => {
+		const node = inputsRef.current[ targetIdx ];
+		if ( node ) {
+			node.focus();
+		}
+	};
+
+	const handleKeyDown = ( e ) => {
+		const { dataset, selectionStart, value } = e.target;
+		const idx = Number( dataset.idx );
+
+		switch ( e.keyCode ) {
+			case KEY_CODE_LEFT:
+			case KEY_CODE_BACKSPACE:
+				if ( selectionStart === 0 ) {
+					maybeMoveFocus( idx - 1 );
+				}
+				break;
+
+			case KEY_CODE_RIGHT:
+				if ( selectionStart === 1 || ! value ) {
+					maybeMoveFocus( idx + 1 );
+				}
+				break;
+		}
+	};
+
+	const handleBeforeInput = ( e ) => {
+		cursorRef.current = e.target.selectionStart;
+	};
+
+	const handleInput = ( e ) => {
+		const { value, dataset } = e.target;
+		const idx = Number( dataset.idx );
+		const digit = value.substr( cursorRef.current, 1 ).replace( /\D/, '' );
+
+		if ( digit !== value ) {
+			e.target.value = digit;
+		}
```

Comment:
:nail_care: 
We do value to (stringified) number sanitization here, right? a code commend would be nice

Reply:
Added by https://github.com/woocommerce/google-listings-and-ads/pull/993/commits/f01870ae93068c01117b7342e66732ff45d490cf.